syntax = "proto3";

package bookstore;

service OrderManagement {
  rpc PlaceOrder(OrderRequest) returns (OrderResponse);
  rpc GetOrderDetails(OrderID) returns (OrderDetails);
  rpc GetOrderHistory(UserID) returns (OrderHistory);
}

message OrderRequest {
  string user_id = 1;
  repeated OrderItem items = 2;
}

message OrderItem {
  string title = 1;
  int32 quantity = 2;
}

message OrderResponse {
  string order_id = 1;
}

message OrderID {
  string order_id = 1;
}

message OrderDetails {
  string order_id = 1;
  string user_id = 2;
  repeated OrderItem items = 3;
  string order_date = 4;
}

message UserID {
  string user_id = 1;
}

message OrderHistory {
  repeated OrderSummary orders = 1;
}

message OrderSummary {
  string order_id = 1;
  repeated string titles = 2;
}
